src/bp/nlu/stan/model-repo.ts

```diff
@@ -37,24 +96,23 @@ export default class ModelRepository {
     try {
       mod = JSON.parse(modelBuff.toString())
     } catch (err) {
-      await fse.remove(fpath)
+      await scopedGhost.deleteFile(MODELS_DIR, fname)
     } finally {
       tmpDir.removeCallback()
       return mod
     }
   }
 
-  public async saveModel(model: NLUEngine.Model, password: string): Promise<void> {
-    const { modelDir } = this
-    const modelFileName = this._makeFileName(model.id, password)
-
+  public async saveModel(model: NLUEngine.Model, options: ModelOwnershipOptions = {}): Promise<void | void[]> {
     const serialized = JSON.stringify(model)
+    const modelName = this._makeFileName(model.id, options.appSecret)
+    const scopedGhost = this._getScopedGhostForAppID(options.appId)
 
+    // TODO replace that logic with in-memory streams
```

Comment:
I don't understand this TODO

Reply:
Well use streams instead of writing on the disk and then reading it to create a buffer and then write it back on the ghost. I'm pretty sure we can do all that in memory